admin/modules/config/plugins.php

```diff
@@ -600,7 +600,8 @@
 function get_plugins_list()
 {
 	// Get a list of the plugin files which exist in the plugins directory
-	$dir = @opendir(MYBB_ROOT."inc/plugins/");
+	$dir = is_dir(MYBB_ROOT."inc/plugins/") && is_readable(MYBB_ROOT."inc/plugins/") ? opendir(MYBB_ROOT."inc/plugins/") : null;
```

Comment:
It looks like there are a few places in the source where `opendir()` is called without additional checks nor the `@` operator. What kinds of errors occur with `@opendir()`?